zeus/metric.py

```diff
@@ -288,28 +319,36 @@ def begin_window(self, name: str) -> None:
                 self.update_period,
                 self.prometheus_url,
                 self.job,
+                sync_execution,
             ),
         )
         self.proc.start()
         if not self.proc.is_alive():
             raise RuntimeError(f"Failed to start monitoring process for {name}.")
 
-    def end_window(self, name: str) -> None:
+        self.window_state[name] = MonitoringProcessState(
+            queue=self.queue, proc=self.proc
+        )
+
+    def end_window(self, name: str, sync_execution: bool = False) -> None:
         """End the energy monitoring window.
 
         Args:
             name (str): The unique name of the measurement window. Must match between calls to 'begin_window' and 'end_window'.
+            sync_execution (bool, optional): Whether to execute monitoring synchronously. Defaults to False.
         """
-        if not hasattr(self, "queue") or self.queue is None:
-            raise RuntimeError(
-                "EnergyCumulativeCounter's 'queue' is not initialized. "
-                "Make sure 'begin_window' is called before 'end_window'."
-            )
-        self.queue.put("stop")
-        if self.proc is not None:
-            self.proc.join(timeout=20)
-            if self.proc.is_alive():
-                self.proc.terminate()
+        if name not in self.window_state:
+            raise ValueError(f"No active monitoring process found for '{name}'.")
+
+        state = self.window_state.pop(name)
+        if self.queue is not None:
+            self.queue.put("stop")
+        else:
+            raise RuntimeError("Queue is not initialized")
```

Comment:
`self.queue` can the queue from any random window??? More specifically, it's going to be the queue that belongs to the most recently started window.

This level of quality is completely unacceptable.